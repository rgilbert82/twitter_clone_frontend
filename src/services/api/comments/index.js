export { default as getCommentsAPI } from './getComments.js';
export { default as createCommentAPI } from './createComment.js';
export { default as editCommentAPI } from './editComment.js';
export { default as deleteCommentAPI } from './deleteComment.js';
