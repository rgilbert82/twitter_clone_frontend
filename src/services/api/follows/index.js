export { default as getFollowAPI } from './getFollow.js';
export { default as createFollowAPI } from './createFollow.js';
export { default as deleteFollowAPI } from './deleteFollow.js';
export { default as getUserFollowersAPI } from './getUserFollowers.js';
export { default as getUserFollowingsAPI } from './getUserFollowings.js';
