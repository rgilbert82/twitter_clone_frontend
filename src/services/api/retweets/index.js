export { default as createRetweetAPI } from './createRetweet.js';
export { default as deleteRetweetAPI } from './deleteRetweet.js';
