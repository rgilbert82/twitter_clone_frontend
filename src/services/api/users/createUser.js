import { get_API_URL } from '../../misc';

export default (userData) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/users`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    }

    request.open('POST', path);
    request.setRequestHeader('Content-Type', 'application/json');
    request.send(JSON.stringify(userData));
  });
}
