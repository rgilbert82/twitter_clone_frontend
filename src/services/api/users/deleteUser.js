import { get_API_URL } from '../../misc';

export default (userId) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/users/${userId}`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(request.responseText);
        } else {
          reject(request.responseText);
        }
      }
    }

    request.open('DELETE', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
