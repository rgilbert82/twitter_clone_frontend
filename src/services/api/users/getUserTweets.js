import { get_API_URL } from '../../misc';

export default (userID) => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/users/${userID}/tweets`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    };

    request.open('GET', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
