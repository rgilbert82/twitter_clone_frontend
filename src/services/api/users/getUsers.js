import { get_API_URL } from '../../misc';

export default () => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/users`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    }

    request.open('GET', path);
    request.send();
  });
}
