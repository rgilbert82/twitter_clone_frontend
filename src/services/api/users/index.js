export { default as getUsersAPI } from './getUsers.js';
export { default as getUserAPI } from './getUser.js';
export { default as createUserAPI } from './createUser.js';
export { default as updateUserAPI } from './updateUser.js';
export { default as getUserTweetsAPI } from './getUserTweets.js';
export { default as deleteUserAPI } from './deleteUser.js';
