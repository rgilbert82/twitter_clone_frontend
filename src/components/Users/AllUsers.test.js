jest.mock('../../services/api/users');

import React from 'react';
import { AllUsers } from '.';
import { shallow } from 'enzyme';
import { getUsersAPI } from '../../services/api/users';

describe(AllUsers, () => {
  let component;
  let fetchSpy = jest.spyOn(AllUsers.prototype, 'fetchUsers');

  beforeEach(() => {
    component = shallow(<AllUsers />);
  });

  afterAll(() => {
    fetchSpy.mockClear();
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("calls fetchUsers on load", () => {
    expect(fetchSpy).toHaveBeenCalled();
  });

  it("fetches the users", () => {
    expect.assertions(1);
    return component.instance().fetchUsers()
      .then(() => {
        component.update();
        expect(component.state('users')).toEqual([{"followers": 1, "following": 1, "id": 1, "name": "Ryan", "slug": "rg", "username": "RG"}]);
      });
  });
});
