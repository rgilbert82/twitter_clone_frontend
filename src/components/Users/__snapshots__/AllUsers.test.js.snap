// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class AllUsers extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = {
      users: [],
      usersFeed: [] };


    this.fetchUsers = this.fetchUsers.bind(this);
    this.scrollListener = this.scrollListener.bind(this);
  }

  componentDidMount() {
    this._isMounted = true;

    if (this._isMounted) {
      this.fetchUsers();
      window.addEventListener('scroll', this.scrollListener);
    }
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.scrollListener);
    this._isMounted = false;
  }

  scrollListener() {
    if (window.innerHeight + window.pageYOffset >= document.body.offsetHeight - 2) {
      if (this.state.usersFeed.length < this.state.users.length) {
        this.setState({ usersFeed: this.state.users.slice(0, this.state.usersFeed.length + 15) });
      }
    }
  }

  fetchUsers() {
    return (0, _users.getUsersAPI)().
    then(data => {
      this.setState({ users: data, usersFeed: data.slice(0, 15) });
    }).catch(() => {
      this.props.displayMessage('Oops! Something went wrong loading this page!');
    });
  }

  render() {
    let users = this.state.usersFeed.map(user => {
      return (
        _react2.default.createElement('li', { className: 'list_item', key: user.id, __source: { fileName: _jsxFileName, lineNumber: 51 }, __self: this },
          _react2.default.createElement(_.UserListItem, { user: user, __source: { fileName: _jsxFileName, lineNumber: 52 }, __self: this })));


    });

    return (
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 58 }, __self: this },
        _react2.default.createElement('h1', { className: 'section_header', __source: { fileName: _jsxFileName, lineNumber: 59 }, __self: this }, 'Popular Users'),
        _react2.default.createElement('ul', { __source: { fileName: _jsxFileName, lineNumber: 60 }, __self: this },
          users)));



  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllUsers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="section_header"
>
          Popular Users
</h1>,
        <ul>
          
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Popular Users",
          "className": "section_header",
        },
        "ref": null,
        "rendered": "Popular Users",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="section_header"
>
            Popular Users
</h1>,
          <ul>
            
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Popular Users",
            "className": "section_header",
          },
          "ref": null,
          "rendered": "Popular Users",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
